ui/flow_of_allowing_to_add_models.md

```diff
@@ -0,0 +1,144 @@
+# Guide for adding a "new" button 
+You need to already have an application record and a GUI controller that implements a GUI list view (including templates).
+This guide will tell you how to add a button to the list view that creates new instances of the item your application record represents.\
+In this guide we will implement the actual creation of the record in an EMS provider.\
+If creation of your items doesn't require interaction with an EMS API, you can adjust the process described here to implement it in manageiq core instead.
+
+The guide assumes that 
+* Your application-record is in `app/models/your_item.rb`
+* Your application-record belongs to an EMS. Usually its instances appear in MIQ by being refreshed from the EMS.
+* Your ui controller is in `plugins/manageiq-ui-classic/app/controllers/your_item_controller.rb`
+* Your provider is in `plugins/manageiq-providers-your_provider`
+* A menu for your item's list view is in `plugins/manageiq-ui-classic/app/heplers/application_helpers/toolbar/your_items_center.rb`
+* You have an application record in `plugins/manageiq-providers-your_provider/app/models/manageiq/providers/your-provider/your_item.rb` that extends `app/models/your_item.rb` with STI. 
+
+You will work with the following repositories. 
+* manageiq
+* manageiq-ui-classic 
+* manageiq-api
+* manageiq-schema 
+* manageiq-providers-your_provider (if you implement the creation in an EMS)
+
+## Application Record
+### Base Application Record
+Make sure that the base application-record in `app/models/your_item.rb` has 
+* `self.create_your_item`
+* `self.raw_create_your_item`
+* `self.create_your_item_queue`
+
+If these aren't present, add these methods to the base record by copying them from some other model in `app/models/`.
+
+### Provider Application Record
+Make sure you have an application-record in `plugins/manageiq-providers-your_provider/app/models/manageiq/providers/your-provider/your_item.rb`.
+It should extend `app/models/your_item.rb` with Rails STI. 
+
+In the application-record in your provider, you need to add two entries.
+1. `supports :create`
+2. `def self.raw_create_your_item(ext_management_system, options = {})` 
+
+The method `self.raw_create_your_item` performs the actual work of creating.\  
+The options hash will contain what you get from your JS form.\
+In this method you need to call your EMS's API to create a real-world instance of your item.\
+You need to check if the request went through OK. If not, you need to throw an exception.
+If the request went OK, you will want to schedule a refresh so that the change will be reflected in MIQ DB.
+
+## Adding support for the new operation 
+1. In `app/models/ext_management_system.rb` add `supports_attribute :supports_create_your_item, :child_model => "YourItem"`
+2. In `manageiq-providers-your_provider/app/models/manageiq/providers/your-provider/your_item.rb`, add `supports :create`
```

Comment:
Only need to talk about adding `supports :create` once, [[ref]](https://github.com/ManageIQ/guides/pull/478/files#diff-7b7fe923f33307d7a91073faa2e44c3b61e1c9c60a8e0db971e67ea7e7a960c6R36)